### ISSUE TITLE
hip-related error while building recent CP2K master on LUMI

### ISSUE THREAD

**Issue by gabriele16** [cp2k/cp2k#2869]
Hello,

I am trying to build the latest version of CP2K on LUMI and experienced an error related to hip. I can't tell if it is a problem due to the libraries/modules used or it is related to a feature that was introduced recently (see https://github.com/cp2k/cp2k/commit/337eb80bc75522ee88d994c513b00003debc09d2#diff-222702f28550f55c721d63d3f7a72952c48da6f11e7e6a42ac91d2c70f59d45c ).

We get first a warning in src/offload/offload_runtime.h:
 ```
warning: implicit declaration of function 'hipDeviceSetLimit'; did you mean 'hipDeviceGetLimit'? [-Wimplicit-function-declaration]
  344 |     OFFLOAD_CHECK(hipDeviceSetLimit(hipLimitMallocHeapSize, required_size));
```

And then the error:
```
offload_runtime.h:344:19: error: use of undeclared identifier 'hipDeviceSetLimit'; did you mean 'hipDeviceGetLimit'?
    OFFLOAD_CHECK(hipDeviceSetLimit(hipLimitMallocHeapSize, required_size));
                  ^~~~~~~~~~~~~~~~~
                  hipDeviceGetLimit
```

We did successfully compile and run an older version [commit 9d075a87ba6edda76ea5a496699dead09fac0ad2](https://github.com/cp2k/cp2k/commit/9d075a87ba6edda76ea5a496699dead09fac0ad2).

In case it can be helpful, here's the list of modules that are loaded:
```
Currently Loaded Modules:
  1) libfabric/1.15.2.0                      5) ModuleLabel/label (S)   9) cray-dsmml/0.2.2       13) cray-fftw/3.3.10.3       17) craype-hugepages2M
  2) craype-network-ofi                      6) lumi-tools/23.04  (S)  10) cray-libsci/22.12.1.1  14) cray-python/3.9.13.1     18) gcc/11.2.0
  3) perftools-base/22.12.0                  7) init-lumi/0.2     (S)  11) PrgEnv-gnu/8.3.3       15) rocm/5.2.3               19) cray-mpich/8.1.23
  4) xpmem/2.5.2-2.4_3.47__gd0f7936.shasta   8) craype/2.7.19          12) craype-x86-trento      16) craype-accel-amd-gfx90a
```

Looking at `/opt/rocm-5.2.3/include/hip/` on LUMI it seems that `hipDeviceSetLimit` is defined only for `nvidia_detail/nvidia_hip_runtime_api.h` but not for the AMD architecture, whereas `hipDeviceGetLimit` is found in `hip/amd_detail/hip_prof_str.h`.

The toolchain ran fine (without sirius) and using the latest cosma version built from source

./install_cp2k_toolchain.sh         --enable-cray          --enable-hip=yes         --gpu-ver=Mi250         --with-cosma=/users/toccigab/scratch/CP2K_v0723/cosma/build/install --with-sirius=no  

Arch file from toolchain without -Werror flags.
[local_hip.psmp.txt](https://github.com/cp2k/cp2k/files/12095214/local_hip.psmp.txt).

If anybody has some ideas on what might be going wrong it would be great. Thanks.
Gabriele

**Comment by mtaillefumier** [comment]
Actually the function is defined in hip on all platforms starting from rocm-5.3.0. LUMI system of course. The only way to fix this is to do a if statement.

**Comment by mtaillefumier** [comment]
It is possible to solve this issue by adding `&& (HIP_VERSION_MAJOR >= 5) && (HIP_VERSION_MINOR >= 3)` to 
https://github.com/cp2k/cp2k/blob/65cca6c7165d0c817769edb47a3ccdcda48d696c/src/offload/offload_runtime.h#L341 and recompile cp2k. 
I will create a patch for this but the real solution is to update rocm.

**Comment by mtaillefumier** [comment]
it should be fixed by PR https://github.com/cp2k/cp2k/pull/2870

**Comment by gabriele16** [comment]
This works now, thanks! Indeed, v5.2.3 is the latest rocm version on LUMI..

**Comment by mtaillefumier** [comment]
glad it worked. A proper fix will be merged soon in develop.

**Comment by vdellabalda** [comment]
I tried to recompile with the latest changes in develop. Unfortunately, the change from `&& (HIP_VERSION_MAJOR >= 5) && (HIP_VERSION_MINOR >= 3)` to `&& (HIP_VERSION > 50200000)` does not work for the rocm version on LUMI which is 5.2.3. I managed to compile with `&& (HIP_VERSION >= 50300000)`. Since v5.2.3 is the last patch before the release of 5.3, `&& (HIP_VERSION > 50200003)` should also work, I have not tested this though.

**Comment by oschuett** [comment]
@vdellabalda, thanks for your feedback. This is should now be fixed via #2894.
